power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -145,7 +145,31 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
 
     using OwnedUpdateDataset = std::tuple<std::vector<typename ComponentType::UpdateType>...>;
 
+    struct UpdateCompProperties {
```

Comment:
probably should wait until the refactoring in the follow-up, but I feel like this is more like a class than a struct:
* after construction, no changes can be made to this.
* there are combinations that do not make sense
  * e.g.: `has_id && ids_all_na` is a combination that can never occur, although it contains really valuable information. We should either keep using the separate flags but make it a class invariant, or combine it into a single enum.
* things like the validation can be done, e.g., in the constructor
* it contains additional accessors outside the raw data members. feels like adding more getters would make sense

Reply:
Refactoring agreed. The combinations of certain flags are largely following the original design.